packages/kotti-pagination/src/components/FlexiblePaginator.vue

```diff
@@ -0,0 +1,123 @@
+<template>
+	<div class="inline-container" :style="containerStyle">
+		<li
+			v-if="showFirstPage"
+			v-text="humanReadablePageNumber(0)"
+			:class="paginatorClasses(0, 'page-item--active')"
+			@click="$emit('setPage', 0)"
+		/>
+		<li v-if="showLeftDots" v-text="'...'" class="c-auto-important page-item" />
+		<li
+			v-for="(page, index) in neighborValues"
+			:key="index"
+			:class="paginatorClasses(page, 'page-item--active')"
+			v-text="humanReadablePageNumber(page)"
+			@click="$emit('setPage', page)"
+		/>
+		<li
+			v-if="showRightDots"
+			v-text="'...'"
+			class="c-auto-important page-item"
+		/>
+		<li
+			v-if="showLastPage"
+			v-text="humanReadablePageNumber(maximumPage)"
+			:class="paginatorClasses(maximumPage, 'page-item--active')"
+			@click="$emit('setPage', maximumPage)"
+		/>
+	</div>
+</template>
+
+<script>
+const ADJACENT_MULTIPLIER = 20
+const BASE_NUMBER_WIDTH = 24.5
+const BASE_OFFSET = 3
+const MAX_NUMBER_OF_ELEMENTS = 5
+const PIXEL_MULTIPLIER = 2.8
+
+export default {
+	name: 'FlexiblePaginator',
+	props: {
+		adjacentAmount: { type: Number, required: true },
+		currentPage: { type: Number, required: true },
+		fixedWidth: { type: Boolean, required: true },
+		maximumPage: { type: Number, required: true },
+	},
+	methods: {
+		paginatorClasses(page, className) {
+			return {
+				'page-item': true,
+				[className]: this.currentPage === page,
+			}
+		},
+		humanReadablePageNumber(page) {
+			return page + 1
+		},
+	},
+	computed: {
+		pixelMargin() {
```

Comment:
I think you can just put it in a box and center it, making calculations nearly unnecessary.